### PR TITLE
App 723 cameras

### PR DESCRIPTION
**Pull request by mrloureed**
https://viam.atlassian.net/browse/APP-723

- only add stream if not active
- only remove stream if one is active (don't remove if other components are using)

* I have created an improvement ticket as a followup. Currently we have to reset tab content when switching in base component, which is not great.

### CODE REVIEW

File: web/frontend/src/app.vue [hunk 1]
```diff
@@ -617,16 +605,10 @@ const waitForClientAndStart = async () => {
   }
 };
 
-const handleSelectCamera = async (event: string) => {
-  if (event === '') {
-    await Promise.all(activePreviews.map((preview) => viewCamera(preview, false)));
-    activePreviews = [];
-    return;
-  }
+let baseCameraState = new Map<string, boolean>();
```
Comment (micheal-parks) [inline web/frontend/src/app.vue:608]: Can we declare this up near the top of the file before it's used in any function?

**Review by DTCurrie — APPROVED** [review]
✅

File: web/frontend/cypress/e2e/robot-components/base.cy.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+describe('should interact with base', () => {
```
Comment (micheal-parks) [inline web/frontend/cypress/e2e/robot-components/base.cy.ts:1]: I'd recommend this language instead:
```ts
describe('base', () => {
  it('should be interactive', () => {
```
This language is a bit more idiomatic and makes tests easier to read when cypress is conducting them

File: web/frontend/src/components/base.vue [hunk 3]
```diff
@@ -42,6 +42,19 @@ const speed = ref(200);
 const spinSpeed = ref(90);
 const angle = ref(0);
 
+let videoStreamStates = new Map<string, boolean>();
+const selectCameras = ref('')
+
+const initStreamState = () => {
+  for (var value of filterResources(props.resources, 'rdk', 'component', 'camera')) {
```
Comment (micheal-parks) [inline web/frontend/src/components/base.vue:49]:
```suggestion
  for (const value of filterResources(props.resources, 'rdk', 'component', 'camera')) {
```

File: web/runtime-shared/templates/webappindex.html [hunk 4]
```diff
@@ -1,6 +1,10 @@
-<html>
+<!DOCTYPE html>
+<html lang="en">
   <head>
-    <link rel="icon" type="image/png" href="data:image/png;base64,iVBORw0KGgo=" />
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
```
Comment (micheal-parks) [inline web/runtime-shared/templates/webappindex.html:5]: Yaaaaaaaaay 🎂 🎉

File: web/frontend/src/main.ts [hunk 5]
```diff
@@ -1,10 +1,7 @@
 import { createApp } from 'vue';
 import './rc/control_helpers';
 import '@viamrobotics/prime';
-import '@fontsource/space-mono/400.css';
-import '@fontsource/space-mono/400-italic.css';
-import '@fontsource/space-mono/700.css';
-import '@fontsource/space-mono/700-italic.css';
+import '@fontsource/space-mono';
```
[the remark this answers is not in the document]
Comment (mrloureed) [reply web/frontend/src/main.ts:4]: I'll add them back. I didn't realize they only loaded on need.

**Review by micheal-parks — APPROVED** [review]
🎉

**Pull request merged**